internal/configs/static_scope.go

```diff
@@ -0,0 +1,209 @@
+// Copyright (c) The OpenTofu Authors
+// SPDX-License-Identifier: MPL-2.0
+// Copyright (c) 2023 HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package configs
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/hashicorp/hcl/v2"
+	"github.com/opentofu/opentofu/internal/addrs"
+	"github.com/opentofu/opentofu/internal/didyoumean"
+	"github.com/opentofu/opentofu/internal/lang"
+	"github.com/opentofu/opentofu/internal/tfdiags"
+	"github.com/zclconf/go-cty/cty"
+)
+
+func newStaticScope(eval *StaticEvaluator, ident StaticIdentifier, stack []StaticIdentifier) *lang.Scope {
+	return &lang.Scope{
+		Data:        staticScopeData{eval, ident, stack},
+		ParseRef:    addrs.ParseRef,
+		SourceAddr:  ident.Subject,
+		BaseDir:     ".", // Always current working directory for now. (same as Evaluator.Scope())
+		PureOnly:    false,
+		ConsoleMode: false,
+	}
+}
+
+type staticScopeData struct {
+	eval   *StaticEvaluator
+	source StaticIdentifier
+	stack  []StaticIdentifier
```

Comment:
Is source actually necessary? Shouldn't including it in `stack` be enough and make more sense?

Reply:
Addressed in 4f9bb5b31d